doc/source/whatsnew/v0.25.0.rst

```diff
@@ -21,6 +21,7 @@ Other Enhancements
 
 - :meth:`Timestamp.replace` now supports the ``fold`` argument to disambiguate DST transition times (:issue:`25017`)
 - :meth:`DataFrame.at_time` and :meth:`Series.at_time` now support :meth:`datetime.time` objects with timezones (:issue:`24043`)
+- :meth:`DataFrame.set_index` now works for instances of ``abc.Iterator``, provided their output is of the same length as the calling frame (:issue:`22484`)
```

Comment:
use this issue number, what about the note for the closed issue in the PR itself? #24969

Reply:
does this still close #24969 ?

Reply:
I updated the OP to clarify, and added the this PR as you asked

Reply:
OK, I now see that #24969 is still open, but that was already closed by ...